app/components/Login.react.js

```diff
@@ -52,25 +54,61 @@ class Login extends Component {
         this.buildOauthUrl = this.buildOauthUrl.bind(this);
         this.oauthPopUp = this.oauthPopUp.bind(this);
         this.logIn = this.logIn.bind(this);
-        this.checkAnyUserAccountIsSaved = this.checkAnyUserAccountIsSaved.bind(this);
     }
 
-    componentWillReceiveProps(nextProps) {
-        this.checkAnyUserAccountIsSaved().then(isSaved => {
-            if (isSaved) {
-                window.location.assign(connectorUrl);
-            }
-        });
+    componentDidMount() {
+        const {serverType} = this.state;
+        const interval = setInterval(() => {
+              usernameLogged = cookie.load('db-connector-user');
+              if (usernameLogged) {
+                  if (serverType === ONPREM) {
+                      this.setState({
+                          status: 'authorized',
+                          statusMessage: 'Saving user information...'
+                      });
+                      this.saveDomainToSettings().then(res => {
+                          if (res.status === 200) {
+                              window.location.assign(connectorUrl);
+                          } else {
+                              this.setState({
+                                  status: 'failure',
+                                  statusMessage: res.text()
```

Comment:
`res.text()` is a promise, so you need a .`then(text => this.setState({statusMessage: text, status: 'failure'})` to get the actual text